.github/workflows/update.yml

```diff
@@ -0,0 +1,102 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+name: Update
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout l10n:main
+        uses: actions/checkout@v4
+        with: { ref: main, fetch-depth: 0 }
+      - name: Refresh l10n:update
+        run: git push origin HEAD:update || true
+  update:
+    needs: prepare
+    strategy:
+      max-parallel: 1
+      matrix:
+        ref: [master, beta, release, esr115]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout l10n:update
+        uses: actions/checkout@v4
+        with: { ref: update, path: l10n }
+      - name: Checkout firefox:${{ matrix.ref }}
+        uses: actions/checkout@v4
+        with:
+          repository: mozilla/gecko-dev
+          ref: ${{ matrix.ref }}
+          path: firefox
+          # TODO: These paths should be generated from the l10n.toml files
+          sparse-checkout: |
+            browser/branding/official/locales
+            browser/extensions/formautofill/locales
+            browser/extensions/report-site-issue/locales
+            browser/locales
+            devtools/client/locales
+            devtools/shared/locales
+            devtools/startup/locales
+            dom/locales
+            mobile/android/branding/official/locales
+            mobile/android/locales
+            netwerk/locales
+            security/manager/locales
+            toolkit/locales
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: pip
+          cache-dependency-path: l10n/_scripts/requirements.txt
+      - run: pip install -r l10n/_scripts/requirements.txt
+      - run: >
+          python -m _scripts.update
+          --branch ${{ matrix.ref }}
+          --commit $(cd ../firefox && git rev-parse --short HEAD)
+          --firefox ../firefox
+          --configs browser/locales/l10n.toml mobile/android/locales/l10n.toml
+        working-directory: l10n
+      - name: git config
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+      - run: git add .
```

Comment:
Note to self: this is confusing… I was wondering why you're not adding the files with the commit message in this step, and that's because you have a `.gitignore` rule that doesn't show up in this PR and excluded all hidden files.

Reply:
This is kept separate from the next step to make its conditionality a bit clearer. The update script won't always change things, so we need
```sh
git diff-index --quiet HEAD
```
to tell us if the `git add` actually added anything.